packages/Search/src/DTK_DistributedSearchTree_def.hpp

```diff
@@ -0,0 +1,52 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#ifndef DTK_DISTRIBUTED_SEARCH_TREE_DEF_HPP
+#define DTK_DISTRIBUTED_SEARCH_TREE_DEF_HPP
+
+#include <details/DTK_DetailsBox.hpp>
+
+#include <Teuchos_Array.hpp>
+#include <Teuchos_CommHelpers.hpp>
+
+namespace DataTransferKit
+{
+
+template <typename DeviceType>
+DistributedSearchTree<DeviceType>::DistributedSearchTree(
+    Teuchos::RCP<Teuchos::Comm<int> const> comm,
+    Kokkos::View<Box const *, DeviceType> bounding_boxes )
+    : _comm( comm )
+    , _local_tree( bounding_boxes )
+{
+    int const comm_rank = _comm->getRank();
+    int const comm_size = _comm->getSize();
+
+    Kokkos::View<Box *, DeviceType> boxes( "rank_bounding_boxes", comm_size );
+    auto boxes_host = Kokkos::create_mirror_view( boxes );
+    boxes_host( comm_rank ) = _local_tree.bounds();
```

Comment:
Is this copy only necessary because of the bug in Trilinos that you reported? If yes, add a comment that we can change this once we update the images. If no, can you add a comment why not.